src/robot.cpp

```diff
@@ -0,0 +1,114 @@
+#include "robot.h"
+
+Robot::Robot(EventQueue* EVqueue) : m_EVqueue(EVqueue)
+{
+
+}
+
+Robot::~Robot()
+{
+    for(int i = 0; i < N_MOTOR; i++)
+    {
+        delete m_motors[i]; //delete[] ?
+    }
+
+    delete m_wheelAngularSpeedVector;
+    delete m_inverseJacobianKinematicsMatrix;
+    delete m_wantedVelocityVector;
+}
+
+
+bool Robot::InitializeMotorbase()
+{
+// CHECK VALUES
+    if(ROBOT_RADIUS <= 0 || WHEEL_RADIUS <= 0 || X_AXIS_ANGLE_MOTORS.size() != N_MOTOR || ADDRESS_MOTORS.size() != N_MOTOR)
+    {
+        return false;
+    }
+
+// INIT MOTORS
+    for(int i = 0; i < N_MOTOR; i++)
+    {
+        m_motors[i] = new Motor(ADDRESS_MOTORS[i], m_EVqueue);
+
+        if(m_motors[i] == nullptr)
+        {
+            return false;
+        }
+    }
+
+// INIT KINEMATICS - MATRIXES
+    m_wheelAngularSpeedVector = new Matrix(N_MOTOR, 1);
+    m_inverseJacobianKinematicsMatrix = new Matrix(N_MOTOR, 3);
+    m_wantedVelocityVector = new Matrix(3, 1);
+
+    if(m_wheelAngularSpeedVector == nullptr || m_inverseJacobianKinematicsMatrix == nullptr || m_wantedVelocityVector == nullptr)
+    {
+        return false;
+    }
+
+// COMPUTE KINEMATICS - INVERSE JACOBIAN MATRIX 
+    for(int i = 0; i < N_MOTOR; i++)
+    {
+        (*m_inverseJacobianKinematicsMatrix)(i,0) = -sin(DEG_TO_RAD * X_AXIS_ANGLE_MOTORS[i]);
+        (*m_inverseJacobianKinematicsMatrix)(i,1) = cos(DEG_TO_RAD * X_AXIS_ANGLE_MOTORS[i]);
+        (*m_inverseJacobianKinematicsMatrix)(i,2) = ROBOT_RADIUS;   
+    }
+
+    (*m_inverseJacobianKinematicsMatrix) *= (1.0 / WHEEL_RADIUS);
+
+    return true;
+}
+
+bool Robot::Move(const int& wanted_distance, const int& wanted_angle, const int& wanted_rotation, 
+    int wanted_speed, const int& wanted_mstep)
+{
+//CHECK VALUES
+    if(wanted_speed < RPM_MINIMUM)
```

Comment:
- [ ] Ternary operator.
- [ ] si wanted_speed = 0 ?

Reply:
Modifié autrement pour refuser le out of range